src/main/java/core/basesyntax/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.dao.FruitTransactionDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitTransactionDao fruitTransactionDao;
+
+    public PurchaseOperation(FruitTransactionDao fruitTransactionDao) {
+        this.fruitTransactionDao = fruitTransactionDao;
+    }
+    @Override
+    public void recount(FruitTransaction transaction) {
+        if (transaction.getQuantity() < 0) {
+            throw new RuntimeException("Quantity can't be less than 0.");
+        }
+        int currentQuantity = fruitTransactionDao.get(transaction.getFruit());
+        currentQuantity -= transaction.getQuantity();
+        fruitTransactionDao.update(transaction.getFruit(), currentQuantity);
```

Comment:
You need to check the result balance before saving it in the storage to ensure it is positive. If the resulting quantity is negative after a purchase, you should throw a RuntimeException to indicate an insufficient quantity for the transaction.